File: inventarioapi/src/main/java/inventario/inventarioapi/model/dto/ConnectionData.java
package inventario.inventarioapi.model.dto;

public record ConnectionData(String url, Long port, String user, String password) {

    public String getJdbcUrl() {
        return String.format("jdbc:sqlserver://%s:%d;databaseName=SCEPRO;encrypt=true;trustServerCertificate=true;sslProtocol=TLSv1.2;", this.url(), this.port());
    }

}

File: inventarioapp/lib/src/models/connection_data.dart
class ConnectionData {
  final String url;
  final int port;
  final String user;
  final String password;

  ConnectionData({
    required this.url,
    required this.port,
    required this.user,
    required this.password,
  });

  Map<String, dynamic> toJson() => {
    'url': url,
    'port': port,
    'user': user,
    'password': password,
  };
}
